src/Symfony/Bundle/FrameworkBundle/Tests/Controller/AbstractControllerTest.php

```diff
@@ -24,6 +24,36 @@ protected function createController()
         return new TestAbstractController();
     }
 
+    /**
+     * This test protects the default subscribed core services against accidental modification.
+     *
+     * @see https://github.com/symfony/symfony/pull/27415
+     */
+    public function testSubscribedServices()
```

Comment:
This doesn't really prevent anything, as it just duplicates the method.
I've no better idea, except maybe an integration test that would rely on autowiring+calling getParameter.
Worth it? As is right now, I'd prefer removing the current test case.

Reply:
I kinda had the same issue - this fix has to go out today and I don't have time to write a really complex test for it.

I do not agree however it doesn't help at all and should be removed. Most specifically it would have prevented the mistake you made in the first place. The test itself may be dumb but it prevents BC breaks in removing or renaming core services that are depended upon, and changing their type. For example if someone changes the `use` for `ContainerInterface` in AbstractController to the PSR version this would catch it. I also specifically wrote it like this instead of an array compare to allow *adding* new subscriptions without failing. Unit tests are meant to prevent regressions and breaking existing code, without limiting new features. That's what it does. In a dumb way, but it works   :P

Reply:
> it would have prevented the mistake you made

I'm not sure, because I would have changed the test accordingly, as that was not a typo buy my intention to target ContainerInterface. What I missed was that this broke autowiring...

Anyway, OK for keeping the test as is.